libs/api/domains/license-service/src/lib/mappers/ehicCardMapper.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@nestjs/common'
+import { getDocument, isDefined } from '@island.is/shared/utils'
+import { format } from 'kennitala'
+import { EhicCardResponse } from '@island.is/clients/license-client'
+
+@Injectable()
+export class EHICCardPayloadMapper implements GenericLicenseMapper {
+  parsePayload(
+    payload?: unknown,
+    locale: Locale = 'is',
+    labels?: GenericLicenseLabels,
+  ): GenericUserLicensePayload | null {
+    const typedPayload = payload as EhicCardResponse
+
+    if (!typedPayload || !typedPayload.expiryDate) return null
+
+    const expired = typedPayload.expiryDate
+      ? !isAfter(new Date(typedPayload.expiryDate?.toISOString()), new Date())
+      : null
+
+    const label = labels?.labels
+    const data: Array<GenericLicenseDataField> = [
+      typedPayload.cardHolderName
+        ? {
+            type: GenericLicenseDataFieldType.Value,
+            label: getLabel('fullName', locale, label),
+            value: typedPayload.cardHolderName ?? '',
+          }
+        : null,
+      typedPayload.cardHolderNationalId
+        ? {
+            type: GenericLicenseDataFieldType.Value,
+            label: getLabel('nationalId', locale, label),
+            value: typedPayload.cardHolderNationalId
+              ? format(typedPayload.cardHolderNationalId)
+              : '',
+          }
+        : null,
+      typedPayload.cardNumber
+        ? {
+            type: GenericLicenseDataFieldType.Value,
+            label: getLabel('cardNumber', locale, label),
+            value: typedPayload.cardNumber,
+          }
+        : null,
+      typedPayload.issued
+        ? {
+            type: GenericLicenseDataFieldType.Value,
+            label: getLabel('publishedDate', locale, label),
+            value: typedPayload.issued.toISOString(),
+          }
+        : null,
+      typedPayload.expiryDate
+        ? {
+            type: GenericLicenseDataFieldType.Value,
+            label: getLabel('validTo', locale, label),
+            value: typedPayload.expiryDate.toISOString(),
+          }
+        : null,
+      {
+        type: GenericLicenseDataFieldType.Value,
+        label: getLabel('publisher', locale, label),
+        value: 'Sjúkratryggingar',
+      },
+    ].filter(isDefined)
+
+    return {
+      data,
+      rawData: JSON.stringify(typedPayload),
+      metadata: {
+        licenseNumber: typedPayload.cardNumber?.toString() ?? '',
+        expired,
+        expireDate: typedPayload.expiryDate.toISOString(),
+        links: [
+          typedPayload.hasTempCard && typedPayload.tempCardPdf
+            ? {
+                label: getLabel('downloadCard', locale, label),
+                value: getDocument(typedPayload.tempCardPdf, 'pdf'),
+                type: GenericUserLicenseMetaLinksType.Download,
+              }
+            : undefined,
+          {
+            label: getLabel('applyForNewCard', locale, label),
+            value: '/umsoknir/evropska-sjukratryggingakortid',
```

Comment:
Is this for this value?:
https://github.com/island-is/island.is/pull/12495/files#diff-531d0a7b989a8548e3a5d5d0076e87ee38c796c382fe1302b343678ea889b53cR105

Reply:
different things, but this shouldn't be hardcoded, will fix l8r